plugin/completion/base_complete.py

```diff
@@ -120,16 +120,20 @@ def init(self, view, settings):
 
         """
         if not view:
+            log.critical(" no view to initialize completer!")
             return
         current_dir = path.dirname(view.file_name())
         search_scope = SearchScope(
             from_folder=current_dir,
             to_folder=settings.project_base_folder)
         self.flags_manager = FlagsManager(
-            use_cmake=settings.generate_flags_with_cmake,
-            flags_update_strategy=settings.cmake_flags_priority,
-            cmake_prefix_paths=settings.cmake_prefix_paths,
+            view=view,
+            settings=settings,
+            compiler_variant=self.compiler_variant,
```

Comment:
But this only happens when init is called. Init is only called if there is no completer readily available. So it will happen only rarely in the case when the translation unit needs to be recreated. To me this makes sense. But do correct me if I am wrong.